bundles/org.openhab.transform.basicprofiles/src/main/java/org/openhab/transform/basicprofiles/internal/profiles/InactivityProfile.java

```diff
@@ -0,0 +1,157 @@
+    private final ProfileCallback callback;
+    private final ScheduledExecutorService scheduler;
+    private final Duration timeout;
+    private final boolean inverted;
+    private final Cleaner.Cleanable cleanable;
+    private boolean closed;
+
+    private @Nullable ScheduledFuture<?> timeoutTask;
+
+    public InactivityProfile(ProfileCallback callback, ProfileContext context) {
+        InactivityProfileConfig config = context.getConfiguration().as(InactivityProfileConfig.class);
+        long mSec = 0;
+        try {
+            QuantityType<?> timeQty = QuantityType.valueOf(config.timeout);
+            if (!Units.SECOND.getDimension().equals(timeQty.getDimension())) {
+                throw new IllegalArgumentException();
+            }
+            mSec = timeQty.toUnit(MetricPrefix.MILLI(Units.SECOND)) instanceof QuantityType<?> mSecQty
+                    ? mSecQty.longValue()
+                    : 0;
+            if (mSec <= 0) {
+                throw new IllegalArgumentException();
+            }
+        } catch (IllegalArgumentException e) {
+            logger.warn("Profile configuration timeout value \"{}\" is invalid", config.timeout);
+        }
+
+        this.callback = callback;
+        this.scheduler = context.getExecutorService();
+        this.timeout = mSec > 0 ? Duration.ofMillis(mSec) : DEFAULT_TIMEOUT;
+        this.inverted = config.inverted != null ? config.inverted : false;
+        this.cleanable = CLEANER.register(this, () -> clean());
+
+        logger.debug("Created(timeout:{}, inverted:{})", timeout, inverted);
+
+        onStateUpdateFromHandler(UnDefType.NULL); // dummy to set initial item state
+    }
+
+    private void onTimeout() {
+        State itemState = OnOffType.from(!inverted);
+        logger.debug("timeout:{} => itemState:{}", timeout, itemState);
+        this.callback.sendUpdate(itemState);
+    }
+
+    @Override
+    public ProfileTypeUID getProfileTypeUID() {
+        return INACTIVITY_UID;
+    }
+
+    @Override
+    public void onStateUpdateFromItem(State itemState) {
+        // do nothing
+    }
+
+    @Override
+    public void onCommandFromItem(Command itemCommand) {
+        // do nothing
+    }
+
+    @Override
+    public void onCommandFromHandler(Command handlerCommand) {
+        Command itemCommand = OnOffType.from(inverted);
+        logger.debug("handlerCommand:{} => itemCommand:{}", handlerCommand, itemCommand);
+        callback.sendCommand(itemCommand);
+        rescheduleTimeoutTask();
+    }
+
+    @Override
+    public void onStateUpdateFromHandler(State handlerState) {
+        State itemState = OnOffType.from(inverted);
+        logger.debug("handlerState:{} => itemState:{}", handlerState, itemState);
+        callback.sendUpdate(itemState);
+        rescheduleTimeoutTask();
+    }
+
+    @Override
+    public void close() throws Exception {
+        // causes immediate clean up instead of waiting on the garbage collector
+        cleanable.clean();
+    }
```

Comment:
Nobody is processing AutoCloseable now, so why keep it?

Reply:
Hmm. Just-in-case. But if you insist I will remove it..

Reply:
It turns out that I cannot remove it because it is required by the JUnit tests

Reply:
Why not just call `cancelTimeoutTask` in the test?

Implementing an interface without a meaningful intention "just in case" isn't a good practice, IMO.
The implementation of AutoCloseable should be removed and the rest of the code refactored accordingly.

Also using `assertXxxx` just to avoid a try-catch and not for the purpose of test coverage is also not a good practice. Once AutoCloseable is removed, if you still deem necessary to have the close() method, then just remove the `throws` part of the method.

Reply:
> Using the assertNotThrows is just a way to call the close method. The essential thing is that close() is called, in order to cancel the scheduled task. Otherwise the scheduled task will continue to run, and thus cause extra call-backs which interfere with the verify counts in subsequent parameterised tests. Yes I could call close() directly but then instead of using assertNotThrows in the unit test I would need to wrap that in a try / catch block.

Is this your reason for keeping AutoCloseable?